itests/eth_filter_test.go

```diff
@@ -748,6 +754,24 @@ func TestMultipleEvents(t *testing.T) {
 	expectedLogIndex := 0
 	var receiptLogs []ethtypes.EthLog
 	for _, receipt := range receipts {
+		require.Equal(t, ethtypes.EthUint64(0x1), receipt.Status)
+
+		// Test GetBlockReceipts
+		blockReceipts, err := client.EthGetBlockReceipts(ctx, ethtypes.EthBlockNumberOrHash{BlockHash: &receipt.BlockHash})
+		require.NoError(t, err)
```

Comment:
the error in CI here is due to the misuse of `LoadExecutedMessages` - you're passing `ts, ts` so it looks messages up in the first `ts` and receipts in the second `ts` and the number of receipts doesn't match the number of messages, so:

```
        	            	failed to load executed messages: mismatching message and receipt counts (3 msgs, 0 rcts)
```

Hence the need to refactor because you have the messages `ts` but not the receipts one because you're only executing messages, not assembling a full new tipset with blocks etc.

Reply:
```go
fromAddr := client.DefaultKey.Address // Get the address to pass
inputData := inputDataFromFrom(ctx, t, client, fromAddr)

// Invoke the sendCoin function to ensure there is a transaction
_, _, err := client.EVM().InvokeContractByFuncName(ctx, client.DefaultKey.Address, contractAddr, "sendCoin(address,uint256)", inputData)
require.NoError(t, err) // Ensure the function call is successful
```

Reply:
I tried to refactor it to
```go	
st, _, err = a.StateManager.TipSetState(ctx, ts)
if err != nil {
	return nil, xerrors.Errorf("failed to load executed messages: %w", err)
}

executedMsgs, err := a.EthEventHandler.EventFilterManager.LoadExecutedMessages(ctx, ts, st)
```

But I am unable to get the right type for st
```
cannot use st (variable of type cid.Cid) as *types.TipSet value in argument to 
```

Could you please help me out with this?

Reply:
see the suggestion above: https://github.com/filecoin-project/lotus/pull/12478#discussion_r1764336196

you need to be passing in the messages and receipts, which you have, not letting `LoadExecutedMessages` figure it out because it's currently designed to take 2 tipsets, but you don't have the second one because you're not actually creating a tipset, you're just executing messages to get receipts; so bypass all of that by refactoring out the need to supply tipsets so you just supply messages and their receipts.

If you apply the diff in my suggestion above you should get closer to this.